test/e2e/garbage_collector.go

```diff
@@ -395,8 +431,8 @@ var _ = framework.KubeDescribe("Garbage collector", func() {
 			return len(rsList.Items) > 0, nil
 
 		})
-		if err == wait.ErrWaitTimeout {
-			err = fmt.Errorf("Failed to wait for the Deployment to create some ReplicaSet: %v", err)
+		if err != nil {
```

Comment:
Let me rephrase this one: in case the error is `wait.ErrWaitTimeout`, you need to overwrite that error with the actual cause, that you didn't find any RSs.